### PR TITLE
Typescript Configuration for developer website

### PR DESCRIPTION
**Pull request by josephgregoryii**
## Dev Dependencies
- Node Types
- React Types
- React-DOM types
- [gatsby-plugin-typegen](https://www.gatsbyjs.com/plugins/gatsby-plugin-typegen/)
- typescript
- ts-loader
- source-map-loader

- typescript-eslint/parser
- typescript-eslint/eslint-plugin

### Notes:
- When linting via command line, there will be a *ton* of linter warnings. This will slowly decrease as we adapt typescript formatting.
- Linter should work as previously intended. 
- We may also turn `on` some linter options at a later point that were needed to be turned `off` to allow the same rules as previous.

### Suggestions:
- I would suggest sandboxing in the typescript environment and see how the linter feels with your editor.

### CODE REVIEW

**Review by clarkmcadoo — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -26,6 +28,7 @@
     "gatsby-plugin-sass": "^4.8.0",
     "gatsby-plugin-sharp": "^3.14.1",
     "gatsby-plugin-sitemap": "^4.4.0",
+    "gatsby-plugin-typegen": "^2.2.4",
```
Comment (rudouglas) [inline package.json:31]: Should these be moved to `devDependencies`?

**Review by rudouglas — CHANGES_REQUESTED** [review]
Just a comment on moving packages to `devDependencies` otherwise this looks awesome

File: package.json [hunk 2]
```diff
@@ -54,6 +57,8 @@
     "react-use": "^15.3.4",
     "sharp": "^0.29.1",
     "snyk": "^1.437.4",
+    "source-map-loader": "^3.0.1",
+    "ts-loader": "^9.2.6",
```
Comment (aswanson-nr) [inline package.json:61]: These should be dev dependencies in a normal react app, please disregard if that's not the case for a Gatsby site.

**Review by aswanson-nr — COMMENTED** [review]
Looks great!! Just saw some deps that should probably be dev dependecies

File: package.json (hunk 1 above)
Comment (josephgregoryii) [reply package.json]: done :)

**Review by aswanson-nr — APPROVED** [review]

**Pull request merged**